src/ngircd/irc-channel.c

```diff
@@ -236,8 +237,56 @@ join_forward(CLIENT *Client, CLIENT *target, CHANNEL *chan,
 	}
 
 	/* tell users in this channel about the new client */
+		#define IMPLEMENT_EXTENDED_JOIN
+	#ifndef IMPLEMENT_EXTENDED_JOIN
 	IRC_WriteStrChannelPrefix(Client, chan, target, false,
 				  "JOIN :%s",  channame);
+	#endif
+	#ifdef IMPLEMENT_EXTENDED_JOIN
+	CLIENT *c;
+	// Pointless alias
+	CHANNEL *Chan;
+	CONN_ID conn;
+	CL2CHAN *cl2chan;
+	Chan = chan;
+	cl2chan = Channel_FirstMember( Chan );
+	while(cl2chan) {
+		c = Channel_GetClient( cl2chan );
+		//if (!Remote) {
```

Comment:
Please remove dead code and don't comment it out if it is no longer needed.
And don't forget to fix the indentation as well! ;-)